Fix gba_cart.ld support for overlays

JoaoBaptMG opened the pull request:
The current linker script uses `__iwram_overlay_end - __iwram_overlay_start` as an offset for `__ewram_lma`, which uses only the maximum size of an IWRAM overlay and is wrong even for only 2 overlays (section `.ewram`'s LMA will overlap with `.iwram1`'s LMA). This fix makes it correctly consider the total size of all overlays when computing the LMA for section `.ewram`.

JoaoBaptMG committed:
Fix gba_cart.ld support for overlays

The current linker script uses `__iwram_overlay_end - __iwram_overlay_start` as an offset for `__ewram_lma`, which uses only the maximum size of an IWRAM overlay and is wrong even for only 2 overlays (section `.ewram`'s LMA will overlap with `.iwram1`'s LMA). This fix makes it correctly consider the total size of all overlays when computing the LMA for section `.ewram`.

LunarLambda commented:
For what it is worth, overlays are also broken for another reason:

The rules for the `.iwram` section match `*(.iwram*)`, which includes `.iwram0`, `.iwram1`, etc...

This means it is actually impossible to put any data into the overlays, since they will all get pulled into the main iwram section instead.

JoaoBaptMG committed:
Remove overlay references on main section

The original section definition for `.iwram` includes all sections that start with `.iwram` on it, which means code supposed to go in IWRAM overlays would be placed on the `.iwram` definition, rendering the overlay feature useless. This commit fix this patch.

JoaoBaptMG commented:
@LunarLambda Thanks for signaling this, I just pushed a commit to solve it.

JoaoBaptMG committed:
Remove unnecessary changes

I admit I copied from my own modifications of the linker script, and it modified way more than necessary... reverting all the modifications now.

LunarLambda commented on gba_cart.ld:
```diff
@@ -160,8 +160,8 @@ SECTIONS
 		__iwram_start__ = ABSOLUTE(.) ;
-		*(.iwram .iwram*)
-		*iwram.*(.text* .data*)
+		*(.iwram)
```
Could you change this to include `.iwram.*` instead of `.iwram*` as before?
Otherwise, linking with libseven will break, which uses `.iwram.{FunctionName}` names.
